qt/python/mantidqt/mantidqt/widgets/sliceviewer/presenters/presenter.py

```diff
@@ -525,3 +541,29 @@ def get_extra_image_info_columns(self, xdata, ydata):
         else:
             extra_cols = {}
         return extra_cols
+
+
+class SliceViewXAxisEditor(XAxisEditor):
+
+    def __init__(self, canvas, axes, slice_viewer):
```

Comment:
This is only minor, but you could probably just pass a function handle to call in `on_ok` rather than the entire sliceviewer presenter object